File: Unsplash/app/src/main/java/com/example/android/unsplash/DetailActivity.java
package com.example.android.unsplash;

import android.app.Activity;
import android.content.Intent;
import android.os.Build;
import android.os.Bundle;
import android.transition.Slide;
import android.transition.Transition;
import android.view.Gravity;
import android.view.View;
import android.view.animation.AnimationUtils;
import android.widget.ImageView;
import android.widget.TextView;
import android.widget.Toolbar;

import com.squareup.picasso.Picasso;

import butterknife.BindInt;
import butterknife.BindView;
import butterknife.ButterKnife;

public class DetailActivity extends Activity {

  public static final String EXTRA_AUTHOR = "EXTRA_AUTHOR";

  @BindView(R.id.toolbar) Toolbar toolbar;
  @BindView(R.id.photo) ImageView photo;
  @BindView(R.id.author) TextView author;
  @BindInt(R.integer.detail_desc_slide_duration) int slideDuration;

  @Override
  protected void onCreate(Bundle savedInstanceState) {
    super.onCreate(savedInstanceState);
    setContentView(R.layout.activity_detail);
    ButterKnife.bind(this);

    Intent intent = getIntent();
    Picasso.get().load(intent.getData()).placeholder(R.color.placeholder).into(photo);
    author.setText(getString(R.string.author, intent.getStringExtra(EXTRA_AUTHOR)));
    toolbar.setNavigationOnClickListener(v -> finishAfterTransition());

    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
      Transition slide = new Slide(Gravity.BOTTOM)
          .addTarget(R.id.description)
          .setInterpolator(AnimationUtils.loadInterpolator(this, android.R.interpolator.linear_out_slow_in))
          .setDuration(slideDuration);
      getWindow().setEnterTransition(slide);
    }
  }
}

File: ActivityTransition/app/src/main/java/com/alexbaryzhikov/activitytransition/DetailActivity.kt
package com.alexbaryzhikov.activitytransition

import android.app.Activity
import android.os.Bundle
import android.widget.ImageView
import android.widget.TextView

class DetailActivity : Activity() {

  override fun onCreate(savedInstanceState: Bundle?) {
    super.onCreate(savedInstanceState)
    setContentView(R.layout.activity_detail)
    setupActivity()
  }

  override fun onNavigateUp(): Boolean {
    onBackPressed()
    return true
  }

  private fun setupActivity() {
    actionBar.setDisplayHomeAsUpEnabled(true)
    val id = intent.getIntExtra("id", -1)
    findViewById<ImageView>(R.id.image)
        .setImageDrawable(resources.obtainTypedArray(R.array.images).getDrawable(id))
    findViewById<TextView>(R.id.title)
        .text = resources.getStringArray(R.array.titles)[id]
    findViewById<TextView>(R.id.description)
        .text = resources.getStringArray(R.array.descriptions)[id]
  }
}
